Data/src/main/java/org/tribuo/data/columnar/RowProcessor.java

```diff
@@ -570,7 +601,145 @@ public RowProcessor<T> copy() {
+         * use {@link #addFeatureProcessor(FeatureProcessor)}.
+         *
+         * @param featureProcessors the processors to add.
+         * @return A copy of this builder with updated state
+         */
+        public Builder<T> setFeatureProcessors(Set<FeatureProcessor> featureProcessors) {
+            this.featureProcessors = featureProcessors;
+            return this;
+        }
+
+        /**
+         * Add a single feature processor to the builder. See {@link #setFeatureProcessors(Set)} for more detail.
+         *
+         * @param featureProcessor the processor to add.
+         * @return A copy of this builder with updated state
+         */
+        public Builder<T> addFeatureProcessor(FeatureProcessor featureProcessor) {
+            featureProcessors.add(featureProcessor);
+            return this;
+        }
+
+        /**
+         * A map from strings (interpreted as regular expressions by {@link Pattern#compile(String)}) to
+         * {@link FieldProcessor}s such that if a field name matches a regular expression, the corresponding FieldProcessor
+         * is used to process it.
+         * <p>
+         * <strong>N.B.</strong> this method overrides all existing values for regex mapping, to non-destructively add
+         * use {@link #addRegexMappingProcessor(String, FieldProcessor)}.
+         *
+         * @param regexMappingProcessors The map from regex strings to FieldProcessors.
+         * @return A copy of this builder with updated state
+         */
+        public Builder<T> setRegexMappingProcessors(Map<String, FieldProcessor> regexMappingProcessors) {
+            this.regexMappingProcessors = regexMappingProcessors;
+            return this;
+        }
+
+        /**
+         * Add a single regex FieldProcessor mapping to the builder. For more detail see {@link #setRegexMappingProcessors(Map)}.
+         *
+         * @return A copy of this builder with updated state
+         */
+        public Builder<T> addRegexMappingProcessor(String regex, FieldProcessor fieldProcessor) {
+            regexMappingProcessors.put(regex, fieldProcessor);
+            return this;
+        }
+
+        /**
+         * Construct the {@link RowProcessor} represented by this builder's state. Throws {@link PropertyException}
+         * if the state is invalid.
+         *
+         * @param fieldProcessors   Thd field processors to use.
+         * @param responseProcessor The response processor to use.
+         * @return The RowProcessor represented by the builder's state
+         */
+        public RowProcessor<T> build(List<FieldProcessor> fieldProcessors, ResponseProcessor<T> responseProcessor) {
+            Map<String, FieldProcessor> fieldProcessorMap = new HashMap<>();
+            for (FieldProcessor fieldProcessor : fieldProcessors) {
+                fieldProcessorMap.put(fieldProcessor.getFieldName(), fieldProcessor);
```

Comment:
This should probably validate that the field processors don't collide, and/or we should expose two endpoints, one for a list which validates, and one which accepts a map.